src/google/protobuf/io/zero_copy_stream_impl_lite.cc

```diff
@@ -150,22 +150,29 @@ int64 ArrayOutputStream::ByteCount() const {
 // ===================================================================
 
 StringOutputStream::StringOutputStream(string* target)
-  : target_(target) {
+  : target_(target), byte_count_((target_ != NULL) ? target_->size() : 0) {
 }
 
 StringOutputStream::~StringOutputStream() {
+  if (target_ != NULL) {
```

Comment:
Some places are constructing the object with nullptr and destroying them. One particular example is LazyStringOutputStream in this file itself. In fact, I had the CHECK earlier, but some protobuf lite tests were failing, so I decided not to make assumption that target would be non-null always.